### PR TITLE
Update contribution guidelines.

### PR DESCRIPTION
**Pull request by jeffmendoza**
Includes a starter contributing.md, contributor ladder, code of conduct.

### CODE REVIEW

**Comment by jeffmendoza** [comment]
@olivekl Also, please acknowledge that I'm putting you down as doc maintainer on the contributor ladder =) Thanks.

**Comment by jeffmendoza** [comment]
Fixes #238

**Comment by olivekl** [comment]
> @olivekl Also, please acknowledge that I'm putting you down as doc maintainer on the contributor ladder =) Thanks.

Ack, and apologies for the delay. I will review soon!

**Review by naveensrinivasan — DISMISSED** [review]
Love it!

File: code-of-conduct.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# Contributor Covenant Code of Conduct
```
Comment (david-a-wheeler) [inline code-of-conduct.md]: Instead of this, grab or link to the OpenSSF CoC: https://openssf.org/community/code-of-conduct/

Comment (david-a-wheeler) [reply code-of-conduct.md]: Maybe add, at the end, a link to the OpenSSF COC?

**Comment by jeffmendoza** [comment]
@olivekl Any thoughts on this? Thanks

**Comment by SecurityCRob** [comment]
Why is a parallel Code of Conduct being developed?  I'm fine with a group self-managing and laying out how they want their collaborators/contributors to "level up", but as an OpenSSF project we should not set up a bifurcated CoC for a workgroup or project.  It all should align point to the Foundation's CoC.  If the OSSF has a gap, or could be improved by some additional ideas, let's amend that and not start up something else please.

**Comment by jeffmendoza** [comment]
@SecurityCRob The CoC meant to be a copy/paste. I grabbed it from another OpenSSF repo. I'll double check that it is in fact the same one and add a link.

**Comment by AevaOnline** [comment]
First of all, I'm very glad to see contribution guidelines getting updated.

I agree with previous comments about the CoC, and would like to add, please do not copy-paste it as this will lead to bit-drift if the original is ever changed. Please simply link to the top-level CoC.

I am concerned about a few aspects of this contributor ladder:
- there is not a clear description of the project's decision-making process, but the "expectations" column for the "leader" role appears to indicate that the leader "makes final decisions on all large proposals" and is, thus, essentially a [benevolent dictator](https://en.wikipedia.org/wiki/Benevolent_dictator_for_life). As a general rule, I (and I believe this to be true org-wide) prefer consensus-driven models with rotating leadership over time.
- the "leader" tier appears to be singular. Is that intentional? If not, I'd like to see this PR outline the path to elect new leaders, define term-length and electorate, etc.
- basically, the contributor guide should describe how someone can "climb" the ladder.

Lastly, I would like to point out that the OpenSSF definition of a sandbox project requires it to have at least two maintainers from different organizations, and the proposed set of maintainers does not appear to meet that bar. While we didn't expressly comment on "partial maintainership" (e.g., having someone only be a maintainer on a subset of a project, like docs) in PR112, it was not my intent to support that scenario in the OpenSSF's project progression policy. 

Is there someone else within the project who can share the highest tier today?

**Comment by jeffmendoza** [comment]
Thanks for the feedback!

> I agree with previous comments about the CoC, and would like to add, please do not copy-paste it as this will lead to bit-drift if the original is ever changed. Please simply link to the top-level CoC.

Updated


> * there is not a clear description of the project's decision-making process, but the "expectations" column for the "leader" role appears to indicate that the leader "makes final decisions on all large proposals" and is, thus, essentially a [benevolent dictator](https://en.wikipedia.org/wiki/Benevolent_dictator_for_life). As a general rule, I (and I believe this to be true org-wide) prefer consensus-driven models with rotating leadership over time.
> * the "leader" tier appears to be singular. Is that intentional? If not, I'd like to see this PR outline the path to elect new leaders, define term-length and electorate, etc.

Pluralized the titles, added some group/consensus description to the leader(s) role.

> * basically, the contributor guide should describe how someone can "climb" the ladder.

Elaborated on the promotion process.

> Is there someone else within the project who can share the highest tier today?

Not currently, hoping that this ladder will help in this area.

**Comment by olivekl** [comment]
> @olivekl Also, please acknowledge that I'm putting you down as doc maintainer on the contributor ladder =) Thanks.

Unfortunately, I don't think I can take on a docs maintainer role. I look forward to contributing when possible, but I don't think I can make a larger commitment at this time.
